src/main.py

```diff
@@ -1,25 +1,74 @@
+import logging
 import os
 from typing import Optional
 
 from fastapi import Depends, FastAPI, Request, Response, status
+import structlog
 
 from constants import JSONLD
 import schemas
 from store import StubCsvStore, StubMetadataStore
 
+from custom_logging import configure_logger, logger
+from structlog.contextvars import bind_contextvars, clear_contextvars
+import time
 # Simple env var flag to allow local browsing of api responses
 # while developing.
 BROWSABLE = os.environ.get("LOCAL_BROWSE_API")
 
 app = FastAPI()
-
-
+configure_logger()
+
+
+@app.middleware("http")
+async def logging_middleware(request: Request, call_next) -> Response:
+    clear_contextvars()
+    # These context vars will be added to all log entries emitted during the request
+    request_id = request.headers.get("request-id")
+    bind_contextvars(request_id=request_id)
+
+    start_time = time.perf_counter_ns()
+
+    try:
+        response = await call_next(request)
+    except Exception as exc:
+        response = Response(status_code=500)
+        structlog.get_logger("api.error").exception("Uncaught exception", exc_info=exc)
+        raise
+    finally:
+        process_time = time.perf_counter_ns() - start_time
+        status_code = response.status_code
+
+    log_message = "Request completed"
+    log_context = {
```

Comment:
so the way dp logging standards work is that anything applicaiton specific goes into the "data" field and these logging context items are going into the root of the dict being logged. So you'll want _something_ like:

```
log_context = {
        "data": {
               "path": request.url.path,
               "method": request.method,
               "status_code": status_code,
               "process_time": process_time / 10 ** 9,  # Convert to seconds
    }
}
```

to get these fields at the correct level.

the _tricky_ bit here is to make sure that if I do eg `log.info("something happened", data={"foo": "bar"})` then the key:value of {"foo": "bar") gets  _added_ to the data dict rather than replacing it.